core/src/main/java/com/linecorp/armeria/common/metric/MeterRegistryUtil.java

```diff
@@ -0,0 +1,274 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.regex.Pattern;
+
+import com.google.common.base.CaseFormat;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.prometheus.PrometheusMeterRegistry;
+import io.micrometer.core.instrument.stats.quantile.Quantiles;
+import io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles;
+import io.micrometer.core.instrument.util.MeterId;
+
+/**
+ * Provides utilities for accessing {@link MeterRegistry}.
+ */
+public final class MeterRegistryUtil {
+
+    private static final Pattern PROMETHEUS_SANITIZE_PREFIX_PATTERN = Pattern.compile("^[^a-zA-Z_]");
+    private static final Pattern PROMETHEUS_SANITIZE_BODY_PATTERN = Pattern.compile("[^a-zA-Z0-9_]");
+    private static final Pattern ALPHANUM_ONLY_PATTERN = Pattern.compile("^[a-zA-Z0-9]+$");
+
+    private static final double[] DEFAULT_QUANTILES = { 0.5, 0.75, 0.95, 0.98, 0.99, 0.999, 1.0 };
+
+    /**
+     * Returns a newly-created {@link Meter} name.
+     */
+    public static String name(MeterRegistry registry, String... name) {
+        return name(registry, MeterUnit.NONE, name);
+    }
+
+    /**
+     * Returns a newly-created {@link Meter} name.
+     */
+    public static String name(MeterRegistry registry, Iterable<String> name) {
+        return name(registry, MeterUnit.NONE, name);
+    }
+
+    /**
+     * Returns a newly-created {@link Meter} name by concatenating the name of the specified {@link MeterId}
+     * and the specified name parts.
+     */
+    public static String name(MeterRegistry registry, MeterUnit unit, MeterId id, String... name) {
```

Comment:
Just meant `name(registry, unit, id, ImmutableList.copyOf(name)`. It avoids the duplicating of the logic `ImmutableList.<String>builder()...` at the expense of an extra copy.